tests/integrationv2/test_external_psk.py

```diff
@@ -0,0 +1,294 @@
+                                       b"bf1b0070ad3c4988", b"83c9367c09a2be78", b"5abc55cd226097a3",
+                                       b"a982117283f82a03", b"a143efd3ff5dd36d", b"64e861be7fd61d28",
+                                       b"27db279cce145077", b"d454a3664d4e6da4", b"d29ee03725a6a4da",
+                                       b"fcd0fc67d2aea705", b"29513e3da2677fa5", b"906c5b3f7d8f92f2",
+                                       b"28bda40dda721470", b"f9fbf297b5aea617", b"646fac5c03272e97",
+                                       b"0727c621a79141ef", b"5f7de6505e5bfbc3", b"88e9334369409393",
+                                       b"4ae4d357" ]
+
+
+class Outcome(enum.Enum):
+    psk_connection = auto()
+    full_handshake = auto()
+    connection_failed = auto()
+
+S2N_TEST_CASE = namedtuple(
+    'S2N_TEST_CASE', 'cipher certificate s2nc_params s2nd_params outcome')
+
+S2N_TEST_SUITE = [
+    S2N_TEST_CASE(cipher=Ciphers.CHACHA20_POLY1305_SHA256,
+                  certificate=None,
+                  s2nd_params=[ '--psk', s2n_known_value_psk_sha256 ],
+                  s2nc_params=[ '--psk', s2n_known_value_psk_sha256 ],
+                  outcome=Outcome.psk_connection),
+    S2N_TEST_CASE(cipher=Ciphers.AES256_GCM_SHA384,
+                  certificate=None,
+                  s2nd_params=[ '--psk', s2n_known_value_psk_sha384 ],
+                  s2nc_params=[ '--psk', s2n_known_value_psk_sha384 ],
+                  outcome=Outcome.psk_connection),
+    S2N_TEST_CASE(cipher=Ciphers.AES128_GCM_SHA256,
+                  certificate=None,
+                  s2nd_params=[ '--psk', s2n_psk_parameters_sha384, 
+                                '--psk', s2n_known_value_psk_sha256 ],
+                  s2nc_params=[ '--psk', s2n_psk_parameters_sha256, 
+                                '--psk', s2n_known_value_psk_sha256 ],
+                  outcome=Outcome.psk_connection),
+    # Note that the following test case's outcome is `full_handshake` instead of `connection_failed` 
+    # when `certificate` is set to `None` because s2nd uses a default certificate whencle a certificate is not provided.
+    S2N_TEST_CASE(cipher=Ciphers.CHACHA20_POLY1305_SHA256,
+                  certificate=None,
+                  s2nd_params=[ '--psk', s2n_psk_parameters_sha384 ],
+                  s2nc_params=[ '--psk', s2n_psk_parameters_sha256 ],
+                  outcome=Outcome.full_handshake),
+    S2N_TEST_CASE(cipher=Ciphers.AES128_GCM_SHA256,
+                  certificate=Certificates.ECDSA_256,
+                  s2nd_params=[ '--psk', s2n_psk_parameters_sha384 ],
+                  s2nc_params=[ '--psk', s2n_psk_parameters_sha256 ],
+                  outcome=Outcome.full_handshake)
+]
+
+OPENSSL_TEST_CASE = namedtuple(
+    'OPENSSL_TEST_CASE', 'cipher certificate s2n_params openssl_params outcome')
+
+OPENSSL_SERVER_S2NC_TEST_SUITE = [
+    OPENSSL_TEST_CASE(cipher=Ciphers.AES128_GCM_SHA256,
```

Comment:
It might be helpful if you label each of these test cases to explain why it needs to be tested. Otherwise, it's a little tricky to compare / contrast and figure out what you're testing.